core/forkid/forkid.go

```diff
@@ -163,7 +191,7 @@ func newFilter(config *params.ChainConfig, genesis common.Hash, headfn func() ui
 			if sums[i] == id.Hash {
 				// Fork checksum matched, check if a remote future fork block already passed
 				// locally without the local node being aware of it (rule #1a).
-				if id.Next > 0 && head >= id.Next {
+				if id.Next > 0 && (head >= id.Next || (id.Next > timestampThreshold && time >= id.Next)) {
```

Comment:
the head is either block or time, we can just compare it with `id.Next`, isn't it?

like 

```go
if id.Next > 0 && head >= id.Next {
      ....
}
```

Reply:
That breaks test 27 :)

Reply:
Yeah, I get it.

`head` can represent it's a block or a timestamp of *local chain*, it's deterministic. But `id.Next` can refer to a block or a timestamp of *remote chain*, it's unknown. 

E.g. local chain defines all block-based forks without timestamp-based forks. Remote defines the same block-based forks + a timestamp fork as Next.

So we need this extra condition `id.Next > timestampThreshold ` to determine if it's a timestamp or a block number.